File: 3959-check-good-integer/3959-check-good-integer.java
class Solution {
    public boolean checkGoodInteger(int n) {
        int d = 0, s = 0;
        while (n != 0) {
            int r = n % 10;
            n /= 10;
            if (r == 0)
                continue;
            d += r;
            s += (r * r);
        }
        return s >= 50 + d;
    }
}

File: 3959-check-good-integer/3959-check-good-integer.py
class Solution:
    def checkGoodInteger(self, n: int) -> bool:
        d = s = 0
        while n:
            r = n % 10
            n //= 10
            if r == 0:
                continue
            d += r
            s += r ** 2
        return s - d >= 50